domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -161,101 +208,340 @@ class TopicListController @Inject constructor(
+    recentlyPlayerChapterProgress: ChapterProgress,
+    completedChapterProgressList: List<ChapterProgress>,
+    topic: Topic,
+    ongoingStoryListBuilder: OngoingStoryList.Builder
+  ) {
+    val recentlyPlayerChapterSummary: ChapterSummary? =
+      story.chapterList.find { chapterSummary ->
+        recentlyPlayerChapterProgress.explorationId == chapterSummary.explorationId
+      }
+    if (recentlyPlayerChapterSummary != null) {
+      val numberOfDaysPassed = (
+        oppiaClock.getCurrentCalendar().timeInMillis -
+          recentlyPlayerChapterProgress.lastPlayedTimestamp
+        ) / ONE_DAY_IN_MS
+      val promotedStory = createPromotedStory(
+        storyId,
+        topic,
+        completedChapterProgressList.size,
+        story.chapterCount,
+        recentlyPlayerChapterSummary.name,
+        recentlyPlayerChapterSummary.explorationId
+      )
+      if (numberOfDaysPassed < ONE_WEEK_IN_DAYS) {
+        ongoingStoryListBuilder.addRecentStory(promotedStory)
+      } else {
+        ongoingStoryListBuilder.addOlderStory(promotedStory)
+      }
+    }
+  }
+
+  private fun getStartedChapterProgressList(storyProgress: StoryProgress): List<ChapterProgress> {
+    return storyProgress.chapterProgressMap.values
+      .filter { chapterProgress ->
+        chapterProgress.chapterPlayState ==
+          ChapterPlayState.STARTED_NOT_COMPLETED
+      }
+      .sortedByDescending { chapterProgress -> chapterProgress.lastPlayedTimestamp }
+  }
+
+  private fun getCompletedChapterProgressList(storyProgress: StoryProgress): List<ChapterProgress> {
+    return storyProgress.chapterProgressMap.values
+      .filter { chapterProgress ->
+        chapterProgress.chapterPlayState ==
+          ChapterPlayState.COMPLETED
+      }
+      .sortedByDescending { chapterProgress -> chapterProgress.lastPlayedTimestamp }
+  }
+
+  private fun createRecommendedActivityList(
+    topicProgressList: List<TopicProgress>
+  ): RecommendedActivityList {
+    val recommendedActivityListBuilder = RecommendedActivityList.newBuilder()
+
+    if (topicProgressList.isNotEmpty()) {
+      val recommendedStoryBuilder = RecommendedStoryList.newBuilder()
+      if (topicProgressList.size == 1) {
+        recommendedStoryBuilder.addAllSuggestedStory(
+          createRecommendedStoryList(
+            topicProgressList,
+            recommendedActivityListBuilder,
+            recommendedStoryBuilder
+          )
+        )
+        recommendedActivityListBuilder.setRecommendedStoryList(recommendedStoryBuilder)
+
+        if (recommendedStoryBuilder.suggestedStoryCount == 0 &&
+          recommendedStoryBuilder.recentlyPlayedStoryCount == 0 &&
+          recommendedStoryBuilder.olderPlayedStoryCount == 0
+        ) {
+          recommendedActivityListBuilder.comingSoonTopicList = createComingSoonTopicList()
+        }
+      } else {
+        // Add recently played stories or last played stories in RecommendedActivityList.
+        createRecentPlayedStories(
```

Comment:
Should probably be ``populateRecentlyPlayedStories`` based on the context of how it's being used.

Reply:
Renamed it to `populateRecentlyPlayedStories`.